ee/clickhouse/queries/actor_base_query.py

```diff
@@ -67,34 +82,99 @@ def is_aggregating_by_groups(self) -> bool:
         """Override in child class with insight specific logic to determine group aggregation"""
         return False
 
+    @cached_property
+    def should_include_recordings(self) -> bool:
+        """Override in child class with insight specific logic to determine when to include recordings"""
+        return False
+
     def get_actors(
         self,
     ) -> Tuple[Union[QuerySet[Person], QuerySet[Group]], Union[List[SerializedGroup], List[SerializedPerson]]]:
         """ Get actors in data model and dict formats. Builds query and executes """
         query, params = self.actor_query()
         raw_result = sync_execute(query, params)
-        return self.get_actors_from_result(raw_result)
+        actors, serialized_actors = self.get_actors_from_result(raw_result)
+
+        if self.should_include_recordings:
+            serialized_actors = self.add_matched_recordings_to_serialized_actors(serialized_actors, raw_result)
+
+        return actors, serialized_actors
+
+    def query_for_session_ids_with_recordings(self, session_ids: Set[str]) -> Set[str]:
+        """ Filters a list of session_ids to those that actually have recordings """
+        query = """
+        SELECT 
+            distinct session_id
+        FROM session_recording_events
+        WHERE
+            team_id = %(team_id)s
+            and has_full_snapshot = 1
+            and session_id in %(session_ids)s
```

Comment:
Good question. This recordings query should be consistently pretty quick because of the `has_full_snapshot` + `session_id` filters, but the changes to the event query will definitely hurt perf until we materialize the `session_id` and `window_id` properties on the event table. Right now, that's being blocked by the events table migration, but once that completes, we can materialize those columns. 

Either way, it's behind a feature flag, so we can test perf in prod.